From 98c6e30f84590763fc8909de1ca8b96512f77d87 Mon Sep 17 00:00:00 2001
From: Ben Pfaff <blp@nicira.com>
Date: Wed, 9 May 2012 12:21:54 -0700
Subject: [PATCH 03/13] odp-util: Fix parsing of actions encapsulated within
 "sample" actions.

Signed-off-by: Ben Pfaff <blp@nicira.com>
---
 lib/odp-util.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/lib/odp-util.c b/lib/odp-util.c
index ee1c378..aae5cc7 100644
--- a/lib/odp-util.c
+++ b/lib/odp-util.c
@@ -447,7 +447,7 @@ parse_odp_action(const char *s, const struct shash *port_names,
             for (;;) {
                 int retval;
 
-                s += strspn(s, delimiters);
+                n += strspn(s + n, delimiters);
                 if (s[n] == ')') {
                     break;
                 }
@@ -457,7 +457,6 @@ parse_odp_action(const char *s, const struct shash *port_names,
                     return retval;
                 }
                 n += retval;
-
             }
             nl_msg_end_nested(actions, actions_ofs);
             nl_msg_end_nested(actions, sample_ofs);
-- 
1.7.5.4

