From 62732fbf18480ae095e540f3d536cbe017731f20 Mon Sep 17 00:00:00 2001
From: Ethan Jackson <ethan@nicira.com>
Date: Tue, 22 Mar 2011 13:16:24 -0700
Subject: [PATCH 10/13] bridge: Memory leak in port_destroy()

---
 vswitchd/bridge.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/vswitchd/bridge.c b/vswitchd/bridge.c
index 9fb2579..17c18da 100644
--- a/vswitchd/bridge.c
+++ b/vswitchd/bridge.c
@@ -4541,6 +4541,7 @@ port_destroy(struct port *port)
         netdev_monitor_destroy(port->monitor);
         free(port->ifaces);
         bitmap_free(port->trunks);
+        free(port->bond_hash);
         free(port->name);
         free(port);
         bridge_flush(br);
-- 
1.6.3.3

