From 7fa8c6709db0ac416a78896545b0005e35556903 Mon Sep 17 00:00:00 2001
From: Justin Pettit <jpettit@nicira.com>
Date: Fri, 8 Oct 2010 17:15:05 -0700
Subject: [PATCH 22/22] Release Open vSwitch 1.0.2

---
 ChangeLog        |    2 +-
 configure.ac     |    2 +-
 debian/changelog |    4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 6e7217a..ff134d9 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,4 +1,4 @@
-v1.1.0pre1 - 31 Aug 2010
+v1.0.2 - 08 Oct 2010
 ------------------------
     - OpenFlow 1.0 slicing (QoS) functionality
     - Python bindings for configuration database (no write support)
diff --git a/configure.ac b/configure.ac
index 21f3473..ccd4dae 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 AC_PREREQ(2.64)
-AC_INIT(openvswitch, 1.1.0pre1, ovs-bugs@openvswitch.org)
+AC_INIT(openvswitch, 1.0.2, ovs-bugs@openvswitch.org)
 NX_BUILDNR
 AC_CONFIG_SRCDIR([datapath/datapath.c])
 AC_CONFIG_MACRO_DIR([m4])
diff --git a/debian/changelog b/debian/changelog
index 89eb517..f426790 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,11 +1,11 @@
-openvswitch (1.1.0pre1) unstable; urgency=low
+openvswitch (1.0.2) unstable; urgency=low
 
   * OpenFlow 1.0 slicing (QoS) functionality
   * Python bindings for configuration database (no write support)
   * Performance and scalability improvements
   * Bug fixes
 
- -- Open vSwitch team <dev@openvswitch.org>  Tue, 31 Aug 2010 23:20:00 +0000
+ -- Open vSwitch team <dev@openvswitch.org>  Tue, 08 Oct 2010 17:12:00 +0000
 
 openvswitch (1.0.1) unstable; urgency=low
 
-- 
1.6.3.3

