diff -r 826ae988a24c Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,6 +5,8 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
 
+MKDIR_P = mkdir -p --
+
 AUTOMAKE_OPTIONS = foreign subdir-objects
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = datapath
diff -r 826ae988a24c configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-AC_PREREQ(2.64)
+AC_PREREQ(2.59)
 AC_INIT(openvswitch, 1.3.0, ovs-bugs@openvswitch.org)
 NX_BUILDNR
 AC_CONFIG_SRCDIR([datapath/datapath.c])
@@ -26,7 +26,6 @@
 AM_PROG_CC_C_O
 AC_PROG_CPP
 AC_PROG_RANLIB
-AC_PROG_MKDIR_P
 AC_PROG_FGREP
 
 AC_ARG_VAR([PERL], [path to Perl interpreter])
@@ -35,7 +34,10 @@
    AC_MSG_ERROR([Perl interpreter not found in $PATH or $PERL.])
 fi
 
-AC_USE_SYSTEM_EXTENSIONS
+AC_GNU_SOURCE
+AC_COMPILE_IFELSE
+AC_RUN_IFELSE
+
 AC_C_BIGENDIAN
 AC_SYS_LARGEFILE
 
diff -r 826ae988a24c m4/openvswitch.m4
--- a/m4/openvswitch.m4
+++ b/m4/openvswitch.m4
@@ -232,9 +232,9 @@
      [ovs_cv_py_[]AS_TR_SH([$1])],
      [ovs_cv_py_[]AS_TR_SH([$1])=no
       if test $HAVE_PYTHON = yes; then
-        AS_ECHO(["running $PYTHON -c 'import $1
-import sys
-sys.exit(0)'..."]) >&AS_MESSAGE_LOG_FD 2>&1
+dnl         AS_ECHO(["running $PYTHON -c 'import $1
+dnl import sys
+dnl sys.exit(0)'..."]) >&AS_MESSAGE_LOG_FD 2>&1
         if $PYTHON -c 'import $1
 import sys
 sys.exit(0)' >&AS_MESSAGE_LOG_FD 2>&1; then
@@ -286,12 +286,12 @@
 mv conftest.$ac_ext conftest2.$ac_ext
 rm -f conftest1.$ac_objext conftest2.$ac_objext conftest$ac_exeext
 ovs_link2='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest1.$ac_ext conftest2.$ac_ext $LIBS >&5'
-AS_IF([_AC_DO_STDERR($ovs_link2) && {
+AS_IF([_AC_EVAL_STDERR($ovs_link2) && {
 	 test -z "$ac_[]_AC_LANG_ABBREV[]_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
 	 test "$cross_compiling" = yes ||
-	 AS_TEST_X([conftest$ac_exeext])
+	 AS_EXECUTABLE_P([conftest$ac_exeext])
        }],
       [$3],
       [echo "$as_me: failed source file 1 of 2 was:" >&5
