# HG changeset patch
# Parent 13de731f4f7e162ddec358fa212b286d245d5dbd

diff -r 13de731f4f7e python/automake.mk
--- a/python/automake.mk
+++ b/python/automake.mk
@@ -34,13 +34,14 @@
 
 if HAVE_PYTHON
 nobase_pkgdata_DATA = $(ovs_pyfiles) $(ovstest_pyfiles)
+ovs-install-data-local: base=$(shell echo $$RPM_BUILD_ROOT)
 ovs-install-data-local:
 	$(MKDIR_P) python/ovs
 	(echo "import os" && \
-	 echo 'PKGDATADIR = os.environ.get("OVS_PKGDATADIR", """$(pkgdatadir)""")' && \
-	 echo 'RUNDIR = os.environ.get("OVS_RUNDIR", """@RUNDIR@""")' && \
+	 echo 'PKGDATADIR = os.environ.get("OVS_PKGDATADIR", """$(subst $(base),,$(pkgdatadir))""")' && \
+	 echo 'RUNDIR = os.environ.get("OVS_RUNDIR", """$(subst $(base),,@RUNDIR@)""")' && \
 	 echo 'LOGDIR = os.environ.get("OVS_LOGDIR", """@LOGDIR@""")' && \
-	 echo 'BINDIR = os.environ.get("OVS_BINDIR", """$(bindir)""")') \
+	 echo 'BINDIR = os.environ.get("OVS_BINDIR", """$(subst $(base),,$(bindir))""")') \
 		> python/ovs/dirs.py.tmp
 	$(MKDIR_P) $(DESTDIR)$(pkgdatadir)/python/ovs
 	$(INSTALL_DATA) python/ovs/dirs.py.tmp $(DESTDIR)$(pkgdatadir)/python/ovs/dirs.py
