diff -r 7bd2342ff7b8 utilities/ovs-ctl.in
--- a/utilities/ovs-ctl.in
+++ b/utilities/ovs-ctl.in
@@ -177,7 +177,7 @@
 
 	# Start ovsdb-server.
 	set ovsdb-server "$DB_FILE"
-	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:ERR -vANY:FILE:INFO
+	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:INFO -vANY:FILE:EMER
 	set "$@" --remote=punix:"$DB_SOCK"
 	set "$@" --remote=db:Open_vSwitch,manager_options
 	set "$@" --private-key=db:SSL,private_key
@@ -205,7 +205,7 @@
 
 	# Start ovs-vswitchd.
 	set ovs-vswitchd unix:"$DB_SOCK"
-	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:ERR -vANY:FILE:INFO
+	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:INFO -vANY:FILE:EMER
 	if test X"$MLOCKALL" != Xno; then
 	    set "$@" --mlockall
 	fi
@@ -216,7 +216,7 @@
 	log_success_msg "ovs-brcompatd is already running"
     elif test X"$BRCOMPAT" = Xyes; then
         set ovs-brcompatd
-	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:ERR -vANY:FILE:INFO
+	set "$@" -vANY:CONSOLE:EMER -vANY:SYSLOG:EMER -vANY:FILE:EMER
 	start_daemon "$OVS_BRCOMPATD_PRIORITY" "$@"
     fi
 }
