Clear stale configuration keys on boot.

diff -r 655d69fcd68a xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Wed Sep 30 11:17:45 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Wed Sep 30 11:38:22 2009 +0100
@@ -264,6 +264,12 @@
     if [ ! -e "$VSWITCHD_CONF" ]; then
         warning "$VSWITCHD_CONF does not exist"
         action "Creating empty $VSWITCHD_CONF" touch "$VSWITCHD_CONF"
+    else
+	/usr/bin/ovs-cfg-mod '-vANY:console:emer' -F "$VSWITCHD_CONF" \
+	    '--del-match=bridge.*' \
+	    '--del-match=port.*' \
+	    '--del-match=bonding.*' \
+	    '--del-match=iface.*'
     fi
 
     start_vswitchd
