diff -r 3afb3deb44db xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Fri Oct 02 14:33:43 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Fri Oct 02 14:43:54 2009 +0100
@@ -265,11 +265,12 @@
         warning "$VSWITCHD_CONF does not exist"
         action "Creating empty $VSWITCHD_CONF" touch "$VSWITCHD_CONF"
     else
-	/usr/bin/ovs-cfg-mod '-vANY:console:emer' -F "$VSWITCHD_CONF" \
-	    '--del-match=bridge.*' \
-	    '--del-match=port.*' \
-	    '--del-match=bonding.*' \
-	    '--del-match=iface.*'
+        /usr/bin/ovs-cfg-mod '-vANY:console:emer' -F "$VSWITCHD_CONF" \
+            '--del-match=bridge.*' \
+            '--del-match=port.*' \
+            '--del-match=bonding.*' \
+            '--del-match=iface.*' \
+            '--del-match=vlan.*'
     fi
 
     start_vswitchd
