diff -r 987e85685736 xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Wed Oct 14 11:24:08 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Wed Oct 14 11:29:25 2009 +0100
@@ -319,9 +319,12 @@
         ;;
     status)
         status -p ovs-vswitchd.pid ovs-vswitchd
-        if [ "$ENABLE_BRCOMPAT" = "y" ] ; then
+        rc=$?
+        if [ $rc -eq 0 ] && [ "$ENABLE_BRCOMPAT" = "y" ] ; then
             status -p ovs-brcompatd.pid ovs-brcompatd
+            rc=$?
         fi
+        exit $rc
         ;;
     version)
         /usr/sbin/ovs-vswitchd -V
