diff -r 727145ddf6f3 xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Wed Sep 30 11:44:52 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Thu Oct 01 09:47:48 2009 +0100
@@ -89,7 +89,7 @@
         action "Inserting llc module" modprobe llc
         action "Inserting openvswitch module" modprobe openvswitch_mod
     fi
-    if [ "$ENABLE_BRCOMPATD" = "y" ] && [ -n "$BRCOMPATD_PIDFILE" ] && ! lsmod | grep -q "brcompat_mod"; then
+    if [ "$ENABLE_BRCOMPAT" = "y" ] && [ -n "$BRCOMPATD_PIDFILE" ] && ! lsmod | grep -q "brcompat_mod"; then
         action "Inserting brcompat module" modprobe brcompat_mod
     fi
 }
@@ -273,7 +273,7 @@
     fi
 
     start_vswitchd
-    if [ "${ENABLE_BRCOMPATD}" = "y" ] ; then
+    if [ "${ENABLE_BRCOMPAT}" = "y" ] ; then
 	start_brcompatd
     fi
     reload_vswitchd  # ensures ovs-vswitchd has fully read config file.
