diff -r a6e76d551399 xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Thu Oct 01 17:26:46 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Fri Oct 02 11:28:03 2009 +0100
@@ -274,7 +274,7 @@
 
     start_vswitchd
     if [ "${ENABLE_BRCOMPAT}" = "y" ] ; then
-	start_brcompatd
+        start_brcompatd
     fi
     reload_vswitchd  # ensures ovs-vswitchd has fully read config file.
     touch /var/lock/subsys/vswitch
@@ -317,7 +317,9 @@
         ;;
     status)
         status -p ovs-vswitchd.pid ovs-vswitchd
-        status -p ovs-brcompatd.pid ovs-brcompatd
+        if [ "$ENABLE_BRCOMPAT" = "y" ] ; then
+            status -p ovs-brcompatd.pid ovs-brcompatd
+        fi
         ;;
     version)
         /usr/sbin/ovs-vswitchd -V
