# HG changeset patch
# Parent 1d17aa356a2cfa62eb50981c2bbb7f03d3f2802b

diff -r 1d17aa356a2c utilities/ovs-ctl.in
--- a/utilities/ovs-ctl.in	Fri Mar 30 09:34:24 2012 +0100
+++ b/utilities/ovs-ctl.in	Fri Mar 30 11:09:03 2012 +0100
@@ -32,13 +32,13 @@ done
 
 insert_openvswitch_mod_if_required () {
     # If openvswitch_mod is already loaded then we're done.
-    test -e /sys/module/openvswitch_mod && return 0
+    test -e /sys/module/openvswitch && return 0
 
-    # Load openvswitch_mod.  If that's successful then we're done.
-    action "Inserting openvswitch module" modprobe openvswitch_mod && return 0
+    # Load openvswitch.  If that's successful then we're done.
+    action "Inserting openvswitch module" modprobe openvswitch && return 0
 
     # If the bridge module is loaded, then that might be blocking
-    # openvswitch_mod.  Try to unload it, if there are no bridges.
+    # openvswitch.  Try to unload it, if there are no bridges.
     test -e /sys/module/bridge || return 1
     bridges=`echo /sys/class/net/*/bridge | sed 's,/sys/class/net/,,g;s,/bridge,,g'`
     if test "$bridges" != "*"; then
@@ -47,13 +47,13 @@ insert_openvswitch_mod_if_required () {
     fi
     action "removing bridge module" rmmod bridge || return 1
 
-    # Try loading openvswitch_mod again.
-    action "Inserting openvswitch module" modprobe openvswitch_mod
+    # Try loading openvswitch again.
+    action "Inserting openvswitch module" modprobe openvswitch
 }
 
 insert_brcompat_mod_if_required () {
-    test -e /sys/module/brcompat_mod && return 0
-    action "Inserting brcompat module" modprobe brcompat_mod
+    test -e /sys/module/brcompat && return 0
+    action "Inserting brcompat module" modprobe brcompat
 }
 
 insert_mod_if_required () {
@@ -282,11 +282,11 @@ force_reload_kmod () {
         action "Removing datapath: $dp" ovs-dpctl del-dp "$dp"
     done
 
-    if test -e /sys/module/brcompat_mod; then
-        action "Removing brcompat module" rmmod brcompat_mod
+    if test -e /sys/module/brcompat; then
+        action "Removing brcompat module" rmmod brcompat
     fi
-    if test -e /sys/module/openvswitch_mod; then
-        action "Removing openvswitch module" rmmod openvswitch_mod
+    if test -e /sys/module/openvswitch; then
+        action "Removing openvswitch module" rmmod openvswitch
     fi
 
     start
