commit 7b99db051b2560290a6e1bc123f1b4e760226faa
Author: Ben Pfaff <blp@nicira.com>
Date:   Wed May 5 10:50:38 2010 -0700

    bridge: Fix double-free in sFlow configuration.

diff --git a/vswitchd/bridge.c b/vswitchd/bridge.c
index c883cf5..35267a6 100644
--- a/vswitchd/bridge.c
+++ b/vswitchd/bridge.c
@@ -823,7 +823,7 @@ bridge_reconfigure(const struct ovsrec_open_vswitch *ovs_cfg)
             }
             ofproto_set_sflow(br->ofproto, &oso);
 
-            svec_destroy(&oso.targets);
+            /* Do not destroy oso.targets because it is owned by sflow_cfg. */
         } else {
             ofproto_set_sflow(br->ofproto, NULL);
         }
