diff -r e95868138a10 xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/etc_init.d_vswitch	Wed Sep 09 08:47:00 2009 +0100
@@ -24,7 +24,7 @@
 test -e /etc/sysconfig/vswitch && . /etc/sysconfig/vswitch
 
 # General config variables in /etc/sysconfig/vswitch
-: ${VSWITCH_BASE:=/root/vswitch}
+: ${VSWITCH_BASE:=/usr}
 : ${ENABLE_BRCOMPAT:=y}
 : ${ENABLE_FAKE_PROC_NET:=y}
 : ${FORCE_COREFILES:=y}
diff -r e95868138a10 xenserver/etc_init.d_vswitch-xapi-update
--- a/xenserver/etc_init.d_vswitch-xapi-update	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/etc_init.d_vswitch-xapi-update	Wed Sep 09 08:47:00 2009 +0100
@@ -22,7 +22,7 @@
 . /etc/init.d/functions
 
 test -e /etc/sysconfig/vswitch && . /etc/sysconfig/vswitch
-: ${VSWITCH_BASE:=/root/vswitch}
+: ${VSWITCH_BASE:=/usr}
 : ${VSWITCHD_CONF:=/etc/ovs-vswitchd.conf}
 : ${VSWITCHD_PIDFILE:=/var/run/ovs-vswitchd.pid}
 : ${VSWITCHD_PRIORITY:=-5}
diff -r e95868138a10 xenserver/etc_profile.d_vswitch.sh
--- a/xenserver/etc_profile.d_vswitch.sh	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/etc_profile.d_vswitch.sh	Wed Sep 09 08:47:00 2009 +0100
@@ -4,11 +4,6 @@
 # are permitted in any medium without royalty provided the copyright
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
-
-PATH=/root/vswitch/bin:$PATH
-export PATH
-MANPATH=/root/vswitch/share/man:$MANPATH
-export MANPATH
 
 alias vswitch='service vswitch'
 
diff -r e95868138a10 xenserver/etc_xapi.d_plugins_vswitch-cfg-update
--- a/xenserver/etc_xapi.d_plugins_vswitch-cfg-update	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/etc_xapi.d_plugins_vswitch-cfg-update	Wed Sep 09 08:47:00 2009 +0100
@@ -30,7 +30,7 @@
 import os
 import subprocess
 
-cfg_mod="/root/vswitch/bin/ovs-cfg-mod"
+cfg_mod="/usr/bin/ovs-cfg-mod"
 vswitchd_cfg_filename="/etc/ovs-vswitchd.conf"
 cacert_filename="/etc/ovs-vswitchd.cacert"
 
diff -r e95868138a10 xenserver/etc_xensource_scripts_vif
--- a/xenserver/etc_xensource_scripts_vif	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/etc_xensource_scripts_vif	Wed Sep 09 08:47:00 2009 +0100
@@ -11,7 +11,7 @@
 
 # Keep other-config/ keys in sync with device.ml:vif_udev_keys
 
-cfg_mod="/root/vswitch/bin/ovs-cfg-mod"
+cfg_mod="/usr/bin/ovs-cfg-mod"
 dump_vif_details="/root/vswitch/scripts/dump-vif-details"
 service="/sbin/service"
 
diff -r e95868138a10 xenserver/root_vswitch_scripts_sysconfig.template
--- a/xenserver/root_vswitch_scripts_sysconfig.template	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/root_vswitch_scripts_sysconfig.template	Wed Sep 09 08:47:00 2009 +0100
@@ -8,7 +8,7 @@
 # without warranty of any kind.
 
 # VSWITCH_BASE: Root directory where vswitch binaries are installed
-# VSWITCH_BASE=/root/vswitch/openvswitch/build
+# VSWITCH_BASE=/usr
 
 # ENABLE_BRCOMPAT: If 'y' than emulate linux bridging interfaces
 #    using the brcompat kernel module and ovs-brcompatd daemon
diff -r e95868138a10 xenserver/usr_lib_xsconsole_plugins-base_XSFeatureVSwitch.py
--- a/xenserver/usr_lib_xsconsole_plugins-base_XSFeatureVSwitch.py	Tue Sep 08 13:34:12 2009 -0700
+++ b/xenserver/usr_lib_xsconsole_plugins-base_XSFeatureVSwitch.py	Wed Sep 09 08:47:00 2009 +0100
@@ -15,7 +15,7 @@
 import socket
 import subprocess
 
-cfg_mod="/root/vswitch/bin/ovs-cfg-mod"
+cfg_mod="/usr/bin/ovs-cfg-mod"
 vswitchd_cfg_filename="/etc/ovs-vswitchd.conf"
 
 if __name__ == "__main__":
