diff -r d2a63c1d4544 utilities/ovs-vsctl.in
--- a/utilities/ovs-vsctl.in	Tue Sep 15 12:11:44 2009 +0100
+++ b/utilities/ovs-vsctl.in	Tue Sep 15 12:11:47 2009 +0100
@@ -21,6 +21,7 @@
 import re
 import stat
 import sys
+import socket
 
 argv0 = sys.argv[0]
 
@@ -123,7 +124,9 @@
         s = os.stat(target)
     if not stat.S_ISSOCK(s.st_mode):
         raise Error("%s is not a Unix domain socket, cannot reload" % target)
-    f = open(target, "r+")
+    skt = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    skt.connect(target)
+    f = os.fdopen(skt.fileno(), "r+")
     f.write("vswitchd/reload\n")
     f.flush()
     f.readline()
