diff -r fd9b2bfcfb80 Makefile.am
--- a/Makefile.am	Fri Sep 11 14:47:23 2009 +0100
+++ b/Makefile.am	Fri Sep 11 15:06:21 2009 +0100
@@ -62,6 +62,7 @@
 	         -e 's,[@]VERSION[@],$(VERSION),g' \
 	         -e 's,[@]localstatedir[@],$(localstatedir),g' \
                 -e 's,[@]pkgdatadir[@],$(pkgdatadir),g' \
+                -e 's,[@]sysconfdir[@],$(sysconfdir),g' \
 	     > $@.tmp
 	@if head -n 1 $@.tmp | grep -q '#!'; then \
 	    echo chmod +x $@.tmp; \
diff -r fd9b2bfcfb80 utilities/ovs-vsctl.in
--- a/utilities/ovs-vsctl.in	Fri Sep 11 14:47:23 2009 +0100
+++ b/utilities/ovs-vsctl.in	Fri Sep 11 15:06:21 2009 +0100
@@ -24,7 +24,7 @@
 
 argv0 = sys.argv[0]
 
-DEFAULT_VSWITCHD_CONF = "@localstatedir@/ovs-vswitchd.conf"
+DEFAULT_VSWITCHD_CONF = "@sysconfdir@/ovs-vswitchd.conf"
 VSWITCHD_CONF = DEFAULT_VSWITCHD_CONF
 
 DEFAULT_VSWITCHD_TARGET = "@RUNDIR@/ovs-vswitchd.pid"
