diff -r acc381dfabd0 xenserver/etc_init.d_vswitch
--- a/xenserver/etc_init.d_vswitch	Thu Oct 01 09:47:48 2009 +0100
+++ b/xenserver/etc_init.d_vswitch	Thu Oct 01 09:48:07 2009 +0100
@@ -24,7 +24,7 @@
 test -e /etc/sysconfig/vswitch && . /etc/sysconfig/vswitch
 
 # General config variables in /etc/sysconfig/vswitch
-: ${ENABLE_BRCOMPAT:=n}
+: ${ENABLE_BRCOMPAT:=y}
 : ${ENABLE_FAKE_PROC_NET:=n}
 : ${FORCE_COREFILES:=y}
 
