diff --git a/xenserver/openvswitch-xen.spec b/xenserver/openvswitch-xen.spec
index f8534d6..b32c074 100644
--- a/xenserver/openvswitch-xen.spec
+++ b/xenserver/openvswitch-xen.spec
@@ -71,7 +71,7 @@ Open vSwitch Linux kernel module compiled against kernel version
 %endif
 
 %build
-./configure --prefix=/usr --sysconfdir=/etc --localstatedir=%{_localstatedir} --enable-ssl %{build_number} %{with_linux}
+./configure --prefix=/usr --sysconfdir=/etc --localstatedir=%{_localstatedir} --enable-ssl %{build_number} %{with_linux} --mandir=%{_mandir}
 make %{_smp_mflags}
 
 %install
@@ -428,30 +428,30 @@ exit 0
 /usr/bin/ovsdb-client
 /usr/bin/ovsdb-tool
 /usr/lib/xsconsole/plugins-base/XSFeatureVSwitch.py
-/usr/man/man1/ovsdb-client.1.gz
-/usr/man/man1/ovsdb-server.1.gz
-/usr/man/man1/ovsdb-tool.1.gz
-/usr/man/man5/ovs-vswitchd.conf.db.5.gz
-/usr/man/man8/ovs-appctl.8.gz
-/usr/man/man8/ovs-ctl.8.gz
-/usr/man/man8/ovs-dpctl.8.gz
-/usr/man/man8/ovs-ofctl.8.gz
-/usr/man/man8/ovs-parse-leaks.8.gz
-/usr/man/man1/ovs-pcap.1.gz
-/usr/man/man1/ovs-tcpundump.1.gz
-/usr/man/man8/ovs-vlan-bug-workaround.8.gz
-/usr/man/man8/ovs-vlan-test.8.gz
-/usr/man/man8/ovs-vsctl.8.gz
-/usr/man/man8/ovs-vswitchd.8.gz
-/usr/man/man1/ovs-benchmark.1.gz
-/usr/man/man8/ovs-bugtool.8.gz
-/usr/man/man8/ovs-controller.8.gz
-/usr/man/man8/ovs-pki.8.gz
-/usr/man/man8/ovs-test.8.gz
+%{_mandir}/man1/ovsdb-client.1.gz
+%{_mandir}/man1/ovsdb-server.1.gz
+%{_mandir}/man1/ovsdb-tool.1.gz
+%{_mandir}/man5/ovs-vswitchd.conf.db.5.gz
+%{_mandir}/man8/ovs-appctl.8.gz
+%{_mandir}/man8/ovs-ctl.8.gz
+%{_mandir}/man8/ovs-dpctl.8.gz
+%{_mandir}/man8/ovs-ofctl.8.gz
+%{_mandir}/man8/ovs-parse-leaks.8.gz
+%{_mandir}/man1/ovs-pcap.1.gz
+%{_mandir}/man1/ovs-tcpundump.1.gz
+%{_mandir}/man8/ovs-vlan-bug-workaround.8.gz
+%{_mandir}/man8/ovs-vlan-test.8.gz
+%{_mandir}/man8/ovs-vsctl.8.gz
+%{_mandir}/man8/ovs-vswitchd.8.gz
+%{_mandir}/man1/ovs-benchmark.1.gz
+%{_mandir}/man8/ovs-bugtool.8.gz
+%{_mandir}/man8/ovs-controller.8.gz
+%{_mandir}/man8/ovs-pki.8.gz
+%{_mandir}/man8/ovs-test.8.gz
 /var/lib/openvswitch
 %exclude /usr/lib/xsconsole/plugins-base/*.py[co]
 %exclude /usr/sbin/ovs-brcompatd
-%exclude /usr/man/man8/ovs-brcompatd.8.gz
+%exclude %{_mandir}/man8/ovs-brcompatd.8.gz
 %exclude /usr/share/openvswitch/scripts/*.py[co]
 %exclude /usr/share/openvswitch/python/*.py[co]
 %exclude /usr/share/openvswitch/python/ovs/*.py[co]
