diff -r 9a920e075e14 xenserver/etc_init.d_openvswitch
--- a/xenserver/etc_init.d_openvswitch	Mon Jun 28 14:14:08 2010 -0700
+++ b/xenserver/etc_init.d_openvswitch	Wed Jun 30 12:05:56 2010 +0100
@@ -29,7 +29,11 @@
     vswitch|openvswitch)
         ;;
     bridge)
-        exit 0
+        if [ "$1" = "status" ] ; then
+           exit 1
+        else
+           exit 0
+        fi
         ;;
     *)
         echo "Open vSwitch disabled (/etc/xensource/network.conf is invalid)" >&2
